v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -0,0 +1,647 @@
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, newTaskDefinition.id);
+        assert.deepEqual(usedCliArguments, [true, true]);
+        assert.deepEqual(res.taskArguments, { flag: true });
+      });
+
+      it("should read the (optional )bool value after the parameter", function () {
+        const command = "npx hardhat task --flag false";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, newTaskDefinition.id);
+        assert.deepEqual(usedCliArguments, [true, true, true]);
+        assert.deepEqual(res.taskArguments, { flag: false });
+      });
+
+      it("should convert on the fly the camelCase parameter to kebab-case", function () {
+        // Parameter with name "optionFlag" should be converted on the fly to "--option-flag"
+        const command = "npx hardhat task --option-flag <value>";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, newTaskDefinition.id);
+        assert.deepEqual(usedCliArguments, [true, true, true]);
+        assert.deepEqual(res.taskArguments, { optionFlag: "<value>" });
+      });
+
+      it("should throw because the parameter is not defined", async function () {
+        const command = "npx hardhat task --undefinedParam <value>";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        assert.throws(
+          () => parseTaskAndArguments(cliArguments, usedCliArguments, hre),
+          new HardhatError(
+            HardhatError.ERRORS.ARGUMENTS.UNRECOGNIZED_NAMED_PARAM,
+            {
+              parameter: "--undefinedParam",
+            },
+          ),
+        );
+      });
+
+      it("should throw because the task parameter is declared before the task name", async function () {
+        const command = "npx hardhat --param <paramValue> task";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        assert.throws(
+          () => parseTaskAndArguments(cliArguments, usedCliArguments, hre),
+          new HardhatError(
+            HardhatError.ERRORS.ARGUMENTS.UNRECOGNIZED_NAMED_PARAM,
+            {
+              parameter: "--param",
+            },
+          ),
+        );
+      });
+
+      it("should throw because the task parameter is declared but no value is associated to it", async function () {
+        const command = "npx hardhat task --param";
```

Comment:
I think there are more interesting cases to test that follow under this situation. For example

`npx hardhat task --param --global-param global-param-value`

Reply:
the difference is that there's no value because the next cli arg has been used, and not because there isn't any

Reply:
But should something like this work?

`npx hardhat task --param --global-param global-param-value <value>`

Where `<value>` will be associated to `--param`?
My understanding was that the next argument after a named parameter should be its value

Reply:
> My understanding was that the next argument after a named parameter should be its value

That's correct, it should fail. I just think it's worth testing it, as it's pretty different.